doc/source/user-experience.rst

```diff
@@ -0,0 +1,282 @@
+   fully reproducible. E.g.:
+
+   .. code-block:: bash
+
+          numpy==1.12.1
+          scipy==0.19.0
+          matplotlib==2.0
+
+4. **Use repo2docker to build your Docker image.**
+   .. code-block:: bash
+
+      jupyter-repo2docker <YOUR-GITHUB-REPOSITORY> --image=gcr.io/<PROJECT-NAME>/<IMAGE-NAME>:<TAG> --no-run
+
+   This tells repo2docker to fetch master of the github repository, and use
+   heuristics to build a docker image of it.
+
+  .. note::
+         - The project name should match your google cloud project's name.
+         - Don’t use underscores in your image name. Other than this it can be
+           anything memorable. This is a bug that will be fixed soon.
+         - The tag should be the first 6 characters of the SHA in the GitHub
+           commit for the image to build from.
+
+5. **Push our newly-built Docker image to the cloud.** You can either push this
+   to Docker Hub, or to the gcloud docker repository. Here we’ll push to the
+   gcloud repository::
+
+       gcloud docker -- push gcr.io/<project-name>/<image-name>:<tag>
+
+6. **Edit the JupyterHub configuration to build from this image.** We do this
+   by editing the ``config.yaml`` file that we originally created to include
+   the jupyter hashes. Edit ``config.yaml`` by including these lines in it:
+
+    .. code-block:: bash
+
+          singleuser:
+            image:
+              name: gcr.io/<project-name>/<image-name>
+              tag: <tag>
+
+7. **Tell helm to update JupyterHub to use this configuration.** Using the
+   normal method to `apply the change <#applying-configuration-changes>`_ to
+   the config.
+8. **Restart your notebook if you are already logging in** If you already have
+   a running JupyterHub session, you’ll need to restart it (by stopping and
+   starting your session from the control panel in the top right). New users
+   won’t have to do this.
+9. **Enjoy your new computing environment!** You should now have a live
+   computing environment built off of the Docker image we’ve created.
+
+   .. note::
+      The contents of your GitHub repository might not show up if you have
+      enabled persistent storage. Disable persistent storage if you want them
+      to show up!
+
+Setting environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another way to affect your user's environment is by setting
+`environment variables <https://en.wikipedia.org/wiki/Environment_variable>`_.
+You can set them up in your Docker image too, but it is sometimes
+easier to set them up in your helm chart!
+
+As usual, you would just edit your ``config.yaml`` file and re-apply!
+
+.. code-block:: yaml
+
+   singleuser:
+     extraEnv:
+       EDITOR: "vim"
+
+This will set the environment variable ``EDITOR`` to the value ``vim``. You
+can set any number of static environment variables here as you want.
+
+Users can read the environment variables in their code in various ways.
```

Comment:
I feel like this little end section isn't necessary...it's not our job to tell users how to access environment variables within their respective languages